source/grid/tria.cc

```diff
@@ -8889,6 +8889,26 @@ Triangulation<dim, spacedim>::set_all_manifold_ids_on_boundary (const types::man
 
 
 template <int dim, int spacedim>
+void
+Triangulation<dim, spacedim>::set_all_manifold_ids_on_boundary (const types::boundary_id b_id,
+    const types::manifold_id m_number)
+{
+  unsigned int counter=0;
```

Comment:
You don't actually need a counter, a book is enough. E.g., `bool boundary_found=false`.